### PR TITLE
Reader: Show/hide bylines correctly

### LINKED ISSUE #13432: Site View and "Admin" bylines
Two problems with the Site View and Admin bylines. For example: https://wordpress.com/read/blogs/118534215

screen shot 2017-04-26 at 11 08 47 am

1. We shouldn't show "by admin" in the header. If the name is "Admin" there, we should hide the whole byline here like we do in cards.

2. In the cards, because we're hiding "Admin" and the site name, there's a blank line at the top of every card. That blank line should be omitted. 

So it should look like this:

sv-admin-fixed

### PR DESCRIPTION
**Pull request by mknepprath**
Fixes #13432 - This will hide the byline in the reader header if the author name is 'admin'. This also removes the extra space at the top of the reader cards by not rendering the author name portion of the byline, rather than just the name within the component.

**Example:**
unspecified-1

**Testing Instructions:**
Check some different feeds in the reader - here are a couple:
- Should show byline: http://calypso.localhost:3000/read/blogs/41425786
- Should hide byline: http://calypso.localhost:3000/read/blogs/118534215

Thanks!

### CODE REVIEW

File: client/blocks/reader-post-card/byline.jsx [hunk 1]
```diff
@@ -93,25 +97,24 @@ class PostByline extends React.Component {
...
-						{ shouldDisplayAuthor && showSiteName && ', ' }
-						{ showSiteName && <ReaderSiteStreamLink
-							className="reader-post-card__site reader-post-card__link"
-							feedId={ feedId }
-							siteId={ siteId }
-							post={ post }>
-							{ siteName }
-						</ReaderSiteStreamLink> }
-					</div>
+					{ shouldDisplayAuthor && <div className="reader-post-card__byline-author-site">
```
Comment (blowery) [inline client/blocks/reader-post-card/byline.jsx]: we also use this component in the following and search streams, where we do want to keep the site info even if we suppress the author info. maybe update the check to `( shouldDisplayAuthor || showSiteName ) &&` ?

**Comment by blowery** [comment]
Hi @mknepprath! Thanks for the PR, we really appreciate it. I left one comment on how you're deciding to suppress the byline div. 

Thanks again!

**Comment by mknepprath** [comment]
@blowery Thanks - I think that did the trick! I committed that update, please review. 🙂

**Comment by jancavan** [comment]
I can confirm that "admin" is no longer displayed, but I'll defer to @blowery on reviewing the code.

Although I'm confused why the live site also no longer displays "admin"? https://wordpress.com/read/blogs/118534215, but it still exists on staging.

**Comment by blowery** [comment]
@jancavan likely an issue where `is_single_user_site` gets suppressed outside the proxy... i think @samouri was maybe working on that?

**Comment by blowery** [comment]
@mknepprath thanks again for the contribution!

**Comment by mknepprath** [comment]
@blowery Yayyy! This is my first code merged into Calypso - very exciting, you're welcome! 💯

**Pull request merged**